reducers/chatsReducer.js

```diff
@@ -26,26 +26,31 @@ function chatsReducer(state = chatsInitialState, action) {
             };
         case types.CREATE_USER_IN_CHAT:
             const { id, display_name: displayName, cid } = payload;
-            const { chats } = state;
             const user = {
                 id,
                 displayName
             };
-            // Concat user object to the users array of assoc. chat
-            const cidIndex = chats.findIndex(
-                (item) => item.id === cid
+            const cidIndex = state.chats.findIndex(
+                (chat) => chat.id === cid
             );
             return {
                 ...state,
                 addedUser: user,
                 chats: {
                     ...state.chats,
-                    [chats[cidIndex].users]: [
+                    [state.chats[cidIndex].users]: [
```

Comment:
i think i missed this last time... it's updating the state to be like this:
![Screen Shot 2019-05-19 at 5 53 53 PM](https://user-images.githubusercontent.com/3858380/57990894-c78e8a00-7a5f-11e9-8f75-7c029df824d1.png)

could do something like
```
const chatToUpdate = Object.assign({}, state.chats[cidIndex]);
const newUsers = { users: [...chatToUpdate.users, user] };
const newChat = {...chatToUpdate, ...newUsers};
return {
                ...state,
                addedUser: user,
                chats: {
                    ...state.chats,
                    [cidIndex]: newChat
                }
            };
```

Reply:
Yikes, I knew there was probably something wrong with what he added previously. I might've mucked it up further too, lmao. I'll push up these changes when I get back home!